maskrcnn_benchmark/structures/segmentation_mask.py

```diff
@@ -148,17 +189,26 @@ class SegmentationMask(object):
     This class stores the segmentations for all objects in the image
     """
 
-    def __init__(self, polygons, size, mode=None):
+    def __init__(self, segms, size, mode=None):
         """
         Arguments:
-            polygons: a list of list of lists of numbers. The first
+            segms: three types
+                (1) polygons: a list of list of lists of numbers. The first
                 level of the list correspond to individual instances,
                 the second level to all the polygons that compose the
                 object, and the third level to the polygon coordinates.
+                (2) rles: COCO's run length encoding format, uncompressed or compressed
+                (3) binary masks
+            size: (width, height)
+            mode: 'polygon', 'mask'. if mode is 'mask', convert mask of any format to binary mask
         """
-        assert isinstance(polygons, list)
-
-        self.polygons = [Polygons(p, size, mode) for p in polygons]
+        assert isinstance(segms, list)
+        if type(segms[0]) != list:
```

Comment:
```suggestion
        if type(segms[0]) != list and type(segms[0]) != Polygons:
```
There is an error when I run training on voc/coco with this `SegmentationMask`: 
after calling `target = target.clip_to_image(remove_empty=True)` in `coco.py`, segms[0] will be Polygon, it shouldn't be assigned as 'Mask' in that case.

Reply:
yes thanks